app/Spago/Build.hs

```diff
@@ -69,6 +69,18 @@ test maybeModuleName maybeLimit paths passthroughArgs = do
     moduleName = fromMaybe (Purs.ModuleName "Test.Main") maybeModuleName
     cmd = "node -e \"require('./output/" <> Purs.unModuleName moduleName <> "').main()\""
 
+-- | Run the project: compile and run the Main
+--   (or the provided module name) with node
+run :: Maybe Purs.ModuleName -> Maybe Int -> [Purs.SourcePath] -> [Purs.ExtraArg] -> IO ()
+run maybeModuleName maybeLimit paths passthroughArgs = do
+  build maybeLimit paths passthroughArgs
+  T.shell cmd T.empty >>= \case
+    T.ExitSuccess   -> pure unit
+    T.ExitFailure n -> die $ "Run failed: " <> T.repr n
+  where
+    moduleName = fromMaybe (Purs.ModuleName "Main") maybeModuleName
+    cmd = "node -e \"require('./output/" <> Purs.unModuleName moduleName <> "').main()\""
```

Comment:
Since these commands are doing basically the same thing, we can extract this common code. I.e. something like:

```hs
run :: Maybe Purs.ModuleName -> Maybe Int -> [Purs.SourcePath] -> [Purs.ExtraArg] -> IO ()
run = runWithNode (Purs.ModuleName "Main")

test :: Maybe Purs.ModuleName -> Maybe Int -> [Purs.SourcePath] -> [Purs.ExtraArg] -> IO ()
test = runWithNode (Purs.ModuleName "Test.Main")

-- | Run the project with node: compile and run with the provided ModuleName
--   (or the default one if that's missing)
runWithNode
  :: Purs.ModuleName
  -> Maybe Purs.ModuleName
  -> Maybe Int
  -> [Purs.SourcePath]
  -> [Purs.ExtraArg]
  -> IO ()
runWithNode defaultModuleName maybeModuleName maybeLimit paths passthroughArgs = do
  build maybeLimit paths passthroughArgs
  T.shell cmd T.empty >>= \case
    T.ExitSuccess   -> pure ()
    T.ExitFailure n -> die $ "Running failed, exit code: " <> T.repr n
  where
    moduleName = fromMaybe defaultModuleName maybeModuleName
    cmd = "node -e \"require('./output/" <> Purs.unModuleName moduleName <> "').main()\""

```

Reply:
I'll update it to do that.

Reply:
I realized I needed to add a `Maybe String` and `String argument for the success/failure messages to keep the messages the same throughout both commands. However, I don't know whether those arguments' types should be `String`, which I know is an alias for `[Char]`, or `Text`.

Reply:
As you might have figured from the CI run, all the strings should be `Text`

You can compile the program locally with `stack build --fast` so you don't have to wait for CI

Also you can get a repl with `stack repl` so you can ask for the types of things, e.g. `:t echo` returns `echo :: Text -> IO ()`)